### PR TITLE
LFR-1925 Adding flow types to components

### PR DESCRIPTION
**Pull request by lukeamarsh**
Adding flow types to misc components

### CODE REVIEW

File: packages/core/src/components/Tabs/Tab.js [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+  selected: boolean,
+  className: string,
+  children: React.Node,
+  onClick: Function,
+  onFocus: Function,
+  onBlur: Function,
+}
+
+export default class Tab extends React.Component<Props> {
+  component: any;
```
Comment (Denyer) [inline packages/core/src/components/Tabs/Tab.js]: we should avoid using `any` as much as possible and use specific types, can use `HTMLButtonElement`

File: packages/core/src/components/Tabs/Tab.js [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
 
-  handleClick = e => {
+  handleClick = (e: SyntheticEvent<>) => {
```
Comment (Denyer) [inline packages/core/src/components/Tabs/Tab.js]:
```suggestion
  handleClick = (e: SyntheticEvent<HTMLButtonElement>) => {
```

File: packages/core/src/components/Tabs/Tab.js [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
     const { onClick, value } = this.props;
     onClick(e, parseInt(value, 10));
   };
 
-  handleFocus = e => {
+  handleFocus = (e: SyntheticEvent<>) => {
```
Comment (Denyer) [inline packages/core/src/components/Tabs/Tab.js]:
```suggestion
  handleFocus = (e: SyntheticEvent<HTMLButtonElement>) => {
```

File: packages/core/src/components/Tabs/Tab.js [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
     const { onFocus, value } = this.props;
     onFocus(e, parseInt(value, 10));
   };
 
-  handleBlur = e => {
+  handleBlur = (e: SyntheticEvent<>) => {
```
Comment (Denyer) [inline packages/core/src/components/Tabs/Tab.js]:
```suggestion
  handleBlur = (e: SyntheticEvent<HTMLButtonElement>) => {
```

File: packages/core/src/components/Tabs/Tabs.js [hunk 5]
```diff
@@ -11,17 +12,24 @@ import css from './Tabs.css';
...
+  children: React.Node,
+}
+
+type State = {
+  activeTabIndex: number,
+  focusedTabIndex: ?number,
+}
+
+export default class Tabs extends React.Component<Props, State> {
+  id: any;
```
Comment (Denyer) [inline packages/core/src/components/Tabs/Tabs.js]: Again, can we avoid using `any`

File: packages/core/src/components/ValueIcons/ValueIconThumbsUp.js [hunk 6]
```diff
@@ -1,15 +1,21 @@
...
 
 import defaultcss from './ValueIcon.css';
 import css from './ValueIconThumbsUp.css';
 
-const ValueIconThumbsUp = props => {
+type Props = {
+  className?: string,
+}
+
+
```
Comment (Denyer) [inline packages/core/src/components/ValueIcons/ValueIconThumbsUp.js]: extra new line

File: packages/core/src/components/WindowDimensionsProvider/WindowDimensionsProvider.js [hunk 7]
```diff
@@ -1,15 +1,31 @@
-import React, { createContext, useContext, useState, useEffect } from 'react'
+// @flow
+
+import * as React from 'react'
+import { createContext, useContext, useState, useEffect } from 'react';
```
Comment (dbatten5) [inline packages/core/src/components/WindowDimensionsProvider/WindowDimensionsProvider.js]: dyou need this import too?

**Pull request merged**